src/main/java/redis/embedded/AbstractRedisInstance.java

```diff
@@ -99,4 +96,33 @@ private void tryWaitFor() {
     public List<Integer> ports() {
         return Arrays.asList(port);
     }
+
+    private static class PrintReaderRunnable implements Runnable {
+        private final BufferedReader reader;
+
+        private PrintReaderRunnable(BufferedReader reader) {
+            this.reader = reader;
+        }
+
+        public void run() {
+            try {
+                readLines();
+            } catch(IOException e) {
+                throw new RuntimeException("Failed to determine if Redis Server started", e);
+            } finally {
+                IOUtils.closeQuietly(reader);
+            }
+        }
+
+        public void readLines() throws IOException {
```

Comment:
Good catch! I have [updated the PR](https://github.com/kstyrc/embedded-redis/pull/39/files#diff-ace388731a732c3b434c9f80f9f875faR100)